ACPs/77-reinventing-subnets/README.md

```diff
@@ -32,48 +32,202 @@ With no Elastic Subnets live on Mainnet, it is clear that Permissionless Subnets
+- `nonce` is a strictly increasing number that denotes the latest validator weight update and provides replay protection for this transaction
+- `weight` is the new `weight` of the validator
+
+### New P-Chain Transaction Types
+
+Both before and after this ACP, to create a Subnet, a `CreateSubnetTx` must be issued on the P-Chain. This transaction includes an `Owner` field which defines the key that today can be used to authorize any validator set additions (`AddSubnetValidatorTx`) or removals (`RemoveSubnetValidatorTx`). 
+
+To be a Permissionless Subnet:
+- This `Owner` key must no longer have the ability to modify the Subnet's validator set
+- New transaction types must support modification of a Subnet's validator set via Warp messages.
 
-To provide maximal flexibility for Permissionless Subnets, the BLS multisignature approach in Avalanche Warp Messaging (AWM) is re-used to approve modifications to the Subnet's validator set. Using the `(chainID, address)` pair defined in the `ConvertSubnetTx`, a Warp Message with an [`AddressedCall`](https://github.com/ava-labs/avalanchego/tree/master/vms/platformvm/warp/payload#addressedcall) payload can be constructed.
+The following new transaction types are introduced on the P-Chain to support this functionality.
+- `ConvertSubnetTx`
+- `RegisterSubnetValidatorTx`
+- `SetSubnetValidatorWeightTx`
+- `DisableSubnetValidatorTx`
+- `IncreaseSubnetValidatorWeightTx`
 
-To validate an `AddressedCall` payload in Avalanche Warp Messaging, the `(chainID, address)` pair is used to lookup the validators of `chainID` and verify that the BLS multi-sig includes a quorum (set to 67%) of the Subnet's validator set. The P-Chain will use Warp Messages with `AddressedCall` payloads to support modifications of the Subnet's validator set using this `(chainID, address)` pair (using the `RegisterSubnetValidatorTx` and `SetSubnetValidatorWeightTx` defined later in the specification).
+#### `ConvertSubnetTx`
 
-#### ConvertSubnetTx
+After a `CreateSubnetTx` is issued, a `ConvertSubnetTx` may be issued by the `Owner` to explicitly convert a Subnet from permissioned to permissionless. This transaction will set the `(chainID, address)` pair that will manage the Subnet's validator set going forward.
+
+The `ConvertSubnetTx` specification is:
 
 ```golang
+type PChainOwner struct {
+    // The threshold number of `Addresses` that must provide a signature in order for
+    // the `PChainOwner` to be considered valid.
+    Threshold uint32 `json:"threshold"`
+    // The 20-byte addresses that are allowed to sign to authenticate a `PChainOwner`.
+    // Note: It is required for:
+    //       - len(Addresses) == 0 if `Threshold` is 0.
+    //       - len(Addresses) >= `Threshold`
+    //       - The values in Addresses to be sorted in ascending order.
+    Addresses []ids.ShortID `json:"addresses"`
+}
+
 type SubnetValidator struct {
-    // Must be Ed25519 NodeID
+    // NodeID of this validator
     NodeID ids.NodeID `json:"nodeID"`
```

Comment:
Should this also be a variable length nodeID?

Reply:
I guess that would make future upgrades to support new node ID types less disruptive? Makes sense to me.